app/ui-react/syndesis/src/modules/data/pages/VirtualizationSqlClientPage.tsx

```diff
@@ -51,10 +71,16 @@ export class VirtualizationSqlClientPage extends React.Component<
               <>
                 <HeaderView virtualizationId={virtualizationId} />
                 <VirtualizationNavBar virtualization={virtualization} />
-                <ViewSqlFormAndTable
-                  viewNames={virtualization.serviceViewDefinitions}
-                  virtualizationName={virtualizationId}
-                />
+                <WithViewEditorStates
+                  idPattern={virtualization.serviceVdbName + '*'}
+                >
+                  {({ data, hasData, error }) => (
+                    <ViewSqlFormAndTable
+                      views={getViewDefinitions(data)}
```

Comment:
I'd suggest `data.map((editorState: ViewEditorState) => editorState.viewDefinition)` and remove `getViewDefinitions` altogether.

Reply:
yeah, good point - I amended the commit